distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/AbstractIntegrationTest.java

```diff
@@ -81,38 +82,121 @@
 
     protected static final String KARAF_VERSION = "2.4.3";
 
-    // TODO: Use the Camel AvailablePortFinder.getNextAvailable() test method
-    protected static final String HTTP_PORT = "9081";
+    protected static final String OPENSEARCH_SOURCE_ID = "openSearchSource";
 
-    protected static final String HTTPS_PORT = "9993";
+    protected static final String CSW_SOURCE_ID = "cswSource";
 
-    protected static final String SSH_PORT = "9101";
+    protected static ServerSocket placeHolderSocket;
 
-    protected static final String RMI_SERVER_PORT = "44445";
+    protected static Integer basePort;
 
-    protected static final String RMI_REG_PORT = "1100";
+    protected static final String DDF_HOME_PROPERTY = "ddf.home";
 
-    protected static final String SECURE_ROOT = "https://localhost:";
+    protected static String ddfHome;
 
-    protected static final String SERVICE_ROOT = SECURE_ROOT + HTTPS_PORT + "/services";
+    public enum DynamicPort {
+        BASE_PORT("org.codice.ddf.system.basePort"), HTTP_PORT(
+                "org.codice.ddf.system.httpPort"), HTTPS_PORT(
+                "org.codice.ddf.system.httpsPort"), SSH_PORT, RMI_SERVER_PORT, RMI_REG_PORT;
 
-    protected static final String INSECURE_ROOT = "http://localhost:";
+        private final String systemProperty;
 
-    protected static final String INSECURE_SERVICE_ROOT = INSECURE_ROOT + HTTP_PORT + "/services";
+        DynamicPort() {
+            this.systemProperty = null;
+        }
 
-    protected static final String REST_PATH = SERVICE_ROOT + "/catalog/";
+        DynamicPort(String systemProperty) {
+            this.systemProperty = systemProperty;
+        }
 
-    protected static final String OPENSEARCH_PATH = REST_PATH + "query";
+        String getSystemProperty() {
+            return this.systemProperty;
+        }
 
-    protected static final String CSW_PATH = SERVICE_ROOT + "/csw";
+        public String getPort(Integer basePort) {
+            return String.valueOf(basePort + this.ordinal());
+        }
 
-    protected static final String OPENSEARCH_SOURCE_ID = "openSearchSource";
+        public String getPort() {
+            return String.valueOf(basePort + this.ordinal());
+        }
+    }
 
-    protected static final String CSW_SOURCE_ID = "cswSource";
+    public static class DynamicString {
```

Comment:
Since this is used for URLs exclusively, it would probably be good to give it a less generic name. Something like...`UrlBuilder` or `DynamicUrl` or some such.